differ/mutationDiffer.go

```diff
@@ -90,41 +91,26 @@ type MutationDiffer struct {
 	utils           xdcrUtils.UtilsIface
 }
 
-type JsonBody struct {
-	Value    []byte
-	Cas      uint64
-	Datatype uint8
-}
-
-type JsonMeta struct {
-	Value    []byte
-	Cas      uint64
-	Seqno    uint64
-	Flags    uint32
-	Expiry   uint32
-	Deleted  uint32
-	Datatype uint8
-	Cvcas    uint64
-	CvSrc    hlv.DocumentSourceId
-	CvVer    uint64
-	Pv       hlv.VersionsMap
-	Mv       hlv.VersionsMap
-	Updated  bool
-}
-
-// GocbResult is a wrapper struct that is composed with properties for both get and getMeta results from gocb
-type GocbResult struct {
-	*JsonBody
-	*JsonMeta
-}
-
-func (r *GocbResult) MarshalJSON() ([]byte, error) {
-	if r.JsonBody != nil {
-		return json.Marshal(r.JsonBody)
-	} else if r.JsonMeta != nil {
-		return json.Marshal(r.JsonMeta)
+func (r *GetResult) MarshalJSON() ([]byte, error) {
+	var dataToBeEncoded map[string]interface{} = make(map[string]interface{})
+	if r.GetMetaResult == nil {
+		dataToBeEncoded[base.JsonBody] = r.Value
+	} else {
+		if r.HLV == nil {
+			dataToBeEncoded[base.JsonBody] = r.Value
+			dataToBeEncoded[base.JsonMetadata] = r.GetMetaResult
+		} else {
+			dataToBeEncoded[base.JsonBody] = r.Value
+			dataToBeEncoded[base.JsonMetadata] = r.GetMetaResult
+			dataToBeEncoded[xdcrCrMeta.XATTR_CVCAS_PATH] = r.GetCvCas()
+			dataToBeEncoded[xdcrCrMeta.XATTR_SRC_PATH] = r.GetCvSrc()
+			dataToBeEncoded[xdcrCrMeta.XATTR_VER_PATH] = r.GetCvVer()
+			dataToBeEncoded[xdcrCrMeta.XATTR_PV_PATH] = r.GetPV()
+			dataToBeEncoded[xdcrCrMeta.XATTR_MV_PATH] = r.GetMV()
+			dataToBeEncoded[base.Updated] = r.Updated
```

Comment:
Can you double check you want to compare `base.Updated`?  From goxdcr code base it looks like this is used internally

Reply:
Here I am encoding `base.Updated` just as a additional info to indicate whether the HLV was implicitly constructed on differ side.
There is no comparison done for the field